### PR TITLE
Add testcases to cover the edge scenarios of on-fail clause

### PR DESCRIPTION
**Pull request by mohanvive**

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause-negative-v2.bal [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import ballerina/io;
```
Comment (anupama-pathirage) [inline tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause-negative-v2.bal:1]: Shouldn't we add the license header for the `.bal` files?

Comment (mohanvive) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause-negative-v2.bal:1]: Hmm. That's a good question. If we check the test files in this repo, then we can find .bal files with and without license header. Do we have a standard on this?

Comment (anupama-pathirage) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause-negative-v2.bal:1]: IINM, we didn't have this on old .bal files as we were not able to put comments before the imports on early days. Later that restriction was removed and we continued to have headers for .bal files. 

May be @hasithaa can give more accurate answer.

Comment (mohanvive) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause-negative-v2.bal:1]: Oh, OK. Let me add it.

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**